pkg/binlog/writer/file.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writer
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+
+	"github.com/pingcap/errors"
+	"github.com/siddontang/go/sync2"
+
+	"github.com/pingcap/dm/pkg/log"
+)
+
+// FileWriter is a binlog event writer which writes binlog events to a file.
+type FileWriter struct {
+	cfg *FileWriterConfig
+
+	mu     sync.RWMutex
+	stage  writerStage
+	offset sync2.AtomicInt64
+
+	file *os.File
+}
+
+// FileWriterStatus represents the status of a FileWriter.
+type FileWriterStatus struct {
+	Stage    string `json:"stage"`
+	Filename string `json:"filename"`
+	Offset   int64  `json:"offset"`
+}
+
+// String implements Stringer.String.
+func (s *FileWriterStatus) String() string {
+	data, err := json.Marshal(s)
+	if err != nil {
+		log.Errorf("[FileWriterStatus] marshal status to json error %v", err)
```

Comment:
```suggestion
		log.Errorf("[FileWriterStatus] marshal status %+v to json error %v", s, err)
```

Reply:
if error is not nil, return the error message instead of empty string?

Reply:
changed to return an error message.